WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRow.swift

```diff
@@ -16,20 +20,39 @@ struct ProductRow: View {
     ///
     let accessibilityHint: String
 
-    init(viewModel: ProductRowViewModel, accessibilityHint: String = "") {
+    /// Image processor to resize images in a background thread to avoid blocking the UI
+    /// 
+    ///
+    private var imageProcessor: ImageProcessor {
+        ResizingImageProcessor(
+            referenceSize: .init(width: Layout.productImageSize * scale, height: Layout.productImageSize * scale),
+            mode: .aspectFill)
+    }
+
+    init(multipleSelectionsEnabled: Bool = false,
+         viewModel: ProductRowViewModel,
+         accessibilityHint: String = "") {
+        self.multipleSelectionsEnabled = multipleSelectionsEnabled
         self.viewModel = viewModel
         self.accessibilityHint = accessibilityHint
     }
 
     var body: some View {
         VStack {
             AdaptiveStack(horizontalAlignment: .leading) {
-                HStack(alignment: .top) {
+                HStack(alignment: .center) {
+                    if multipleSelectionsEnabled {
+                        Image(uiImage: viewModel.selectedState.image)
+                            .frame(width: Layout.checkImageSize * scale, height: Layout.checkImageSize * scale)
+                            .foregroundColor(.init(UIColor.brand))
+                    }
+
                     // Product image
                     KFImage.url(viewModel.imageURL)
                         .placeholder {
                             Image(uiImage: .productPlaceholderImage)
                         }
+                        .setProcessor(imageProcessor)
```

Comment:
This is an additional update - the processor is necessary to resize images in a background thread to avoid blocking the UI. In the previous iteration, loading products is laggy if the list contains large images.

Reply:
Nice, TIL!